src/analysis/calculus/local_extr.lean

```diff
@@ -192,7 +192,7 @@ if hf : differentiable_within_at ℝ f s a
 then h.has_fderiv_within_at_eq_zero hf.has_fderiv_within_at hy hy'
 else by { rw fderiv_within_zero_of_not_differentiable_within_at hf, refl }
 
-/-- Fermat's Theorem: the derivative of a function at a local minimum equals zero. -/
+/-- **Fermat's Theorem**: the derivative of a function at a local minimum equals zero. -/
```

Comment:
Same question as above: there are a few more versions of Fermat's theorem below.

Reply:
Yes, but then should not the theorem name plus the following text be included in the bold, to distinguish the different flavors of Fermat's Theorem (local minimum vs local maximum)? An auto-generated index could pick up this text so there are not multiple entries for "Fermat's Theorem" with nothing to distinguish them. If not capture the extra text, then perhaps only the first mention of "Fermat's Theorem" should be in bold--anyone will notice the other versions just following.

Reply:
Note that I'm still against auto-generating indices by regexing for bold markdown text; I would prefer for us to be able to make the docs look however we want them to without worrying about how it will be parsed. In particular, I think bolding more than a few words is not good style.

Reply:
There are 12 lemmas in a row with the docstring "Fermat's Theorem"--it does not make sense for all of them to be bold in my view. My idea for bolding these was to auto-generate an index, so now I am not sure why to do it except for cosmetics

Reply:
I bolded all of the Fermat's theorems